gateway/cert_test.go

```diff
@@ -1732,4 +1807,257 @@ func TestStaticMTLSAPI(t *testing.T) {
+			spec.APIID = "apiID-1"
+			spec.UseMutualTLSAuth = true
+			spec.Proxy.ListenPath = "/static-mtls"
+			spec.ClientCertificates = []string{rootCertID}
+		})
+
+		t.Run("valid client", func(t *testing.T) {
+			validCertClient := GetTLSClient(&clientCert, rootCertPEM)
+			_, _ = ts.Run(t, test.TestCase{
+				Domain: "localhost",
+				Client: validCertClient,
+				Path:   "/static-mtls",
+				Code:   http.StatusOK,
+			})
+		})
+
+		t.Run("invalid client", func(t *testing.T) {
+			_, _, _, invalidClientCert := crypto.GenCertificate(&x509.Certificate{}, false)
+			tlsConfig := GetTLSConfig(&invalidClientCert, nil)
+			tlsConfig.InsecureSkipVerify = false
+			transport := &http.Transport{TLSClientConfig: tlsConfig}
+
+			invalidClient := &http.Client{Transport: transport}
+			u, err := url.Parse(ts.URL)
+
+			req, err := http.NewRequest(http.MethodGet, fmt.Sprintf("localhost:%s/static-mtls", u.Port()), nil)
+			assert.NoError(t, err)
+			_, err = invalidClient.Do(req)
+			assert.Error(t, err)
+		})
+
+	})
+}
+
+func generateRootCertAndKey(tb testing.TB) ([]byte, []byte, error) {
+	tb.Helper()
+	// Generate RSA key pair
+	rootKey, err := rsa.GenerateKey(rand.Reader, 2048)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// Create a template for the root certificate
+	rootCertTemplate := x509.Certificate{
+		SerialNumber: big.NewInt(1),
+		Subject: pkix.Name{
+			Organization:  []string{"Tyk Technologies Ltd"},
+			Country:       []string{"UK"},
+			Province:      []string{"London"},
+			Locality:      []string{"London"},
+			StreetAddress: []string{"Worship Street"},
+		},
+		NotBefore:             time.Now(),
+		NotAfter:              time.Now().AddDate(10, 0, 0), // 10 years
+		KeyUsage:              x509.KeyUsageCertSign | x509.KeyUsageCRLSign,
+		BasicConstraintsValid: true,
+		IsCA:                  true,
+	}
+
+	// Create the root certificate
+	rootCertDER, err := x509.CreateCertificate(rand.Reader, &rootCertTemplate, &rootCertTemplate, &rootKey.PublicKey, rootKey)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// Encode the root certificate to PEM format
+	var rootCertPEM bytes.Buffer
+	_ = pem.Encode(&rootCertPEM, &pem.Block{Type: "CERTIFICATE", Bytes: rootCertDER})
+
+	// Encode the root private key to PEM format
+	var rootKeyPEM bytes.Buffer
+	_ = pem.Encode(&rootKeyPEM, &pem.Block{Type: "RSA PRIVATE KEY", Bytes: x509.MarshalPKCS1PrivateKey(rootKey)})
+
+	return rootCertPEM.Bytes(), rootKeyPEM.Bytes(), nil
+}
+
+func generateServerCertAndKeyPEM(tb testing.TB, rootCertPEM, rootKeyPEM []byte) (*bytes.Buffer, *bytes.Buffer, error) {
+	tb.Helper()
+
+	rootCertBlock, _ := pem.Decode(rootCertPEM)
+	if rootCertBlock == nil || rootCertBlock.Type != "CERTIFICATE" {
+		return nil, nil, fmt.Errorf("failed to decode root certificate PEM")
+	}
+	rootCert, err := x509.ParseCertificate(rootCertBlock.Bytes)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// Decode the root private key
+	rootKeyBlock, _ := pem.Decode(rootKeyPEM)
```

Comment:
The omitted values are errors? Cover with assert.NoError(), or add error checks for swallowed values

Reply:
It's not error, it's the `rest`/`remainder` after decoding PEM block